### PR TITLE
feat: Add Resilience Hub resources

### LINKED ISSUE #6977: AWS Resilience Hub
### Why do you need this resource or plugin? Please describe.
AWS Resilience Hub. AWS Resilience Hub provides actionable recommendations to improve resilience. The resilience assessment also generates code snippets that help you create recovery procedures as AWS Systems Manager documents for your applications, referred to as Standard Operating Procedures (SOPs). AWS Resilience Hub generates a list of recommended Amazon CloudWatch monitors and alarms to help the operator quickly identify any change to the application’s resilience posture once deployed.

https://docs.aws.amazon.com/sdk-for-go/api/service/resiliencehub/

### Describe the solution you'd like
Ideally pull in apps (with tags) and recommendations

### PR DESCRIPTION
**Pull request by amanenk**
#### Summary
closes #6977

### CODE REVIEW

File: plugins/source/k8s/client/client.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 
 	// import all k8s auth options
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/rest"
```
Comment (erezrokah) [inline plugins/source/k8s/client/client.go:16]: The changes in this file seems unrelated to this PR, can we do them in a separate one?
Otherwise the the k8s plugin will get a changelog entry with `feat: Add Resilience Hub resources`

File: website/pages/docs/plugins/sources/k8s/overview.mdx [hunk 2]
```diff
@@ -18,4 +18,30 @@ file (`~/.kube/config`). You can also specify a different configuration by setti
 
 ```bash
 export KUBECONFIG=<PATH_TO_YOUR_CONFIG_FILE>
-```
\ No newline at end of file
+```
+
+If `cloudquery` is running in a pod of the Kubernetes cluster whose information will be synced, the k8s serviceaccount can be used for authentication directly.
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/k8s/overview.mdx:23]: Should be fixed by syncing with `main`

**Review by erezrokah — CHANGES_REQUESTED** [review]
AWS part looks good, can we move the k8s fixes to a separate PR?

**Review by erezrokah — APPROVED** [review]

File: plugins/source/aws/resources/services/resiliencehub/recommendation_templates.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func recommendationTemplates() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_recommendation_templates",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_RecommendationTemplate.html`,
+		Resolver:    fetchRecommendationTemplates,
+		Transform:   transformers.TransformWithStruct(&types.RecommendationTemplate{}, transformers.WithPrimaryKeys("Name")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/recommendation_templates.go:15]: `Name` is the only PK? Should add `account_id` (and maybe `region` too) to avoid PK issues with multiple accounts unless Name is required to be unique globally.

File: plugins/source/aws/resources/services/resiliencehub/alarm_recommendations.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func alarmRecommendations() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_alarm_recommendations",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_AlarmRecommendation.html`,
+		Resolver:    fetchAlarmRecommendations,
+		Transform:   transformers.TransformWithStruct(&types.AlarmRecommendation{}, transformers.WithPrimaryKeys("Name")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/alarm_recommendations.go:15]: Check PK uniqueness

File: plugins/source/aws/resources/services/resiliencehub/app_component_compliances.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func appComponentCompliances() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_app_component_compliances",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_AppComponentCompliance.html`,
+		Resolver:    fetchAppComponentCompliances,
+		Transform:   transformers.TransformWithStruct(&types.AppComponentCompliance{}, transformers.WithPrimaryKeys("AppComponentName")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/app_component_compliances.go:15]: Check PK uniqueness, should add `account_id` and `region`?

File: plugins/source/aws/resources/services/resiliencehub/app_versions.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func appVersions() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_app_versions",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_AppVersionSummary.html`,
+		Resolver:    fetchAppVersions,
+		Transform:   transformers.TransformWithStruct(&types.AppVersionSummary{}, transformers.WithPrimaryKeys("AppVersion")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/app_versions.go:15]: Check PK uniqueness, should add `account_id` and `region`?

File: plugins/source/aws/resources/services/resiliencehub/component_recommendations.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func appComponentRecommendations() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_component_recommendations",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_ComponentRecommendation.html`,
+		Resolver:    fetchComponentRecommendations,
+		Transform:   transformers.TransformWithStruct(&types.ComponentRecommendation{}, transformers.WithPrimaryKeys("AppComponentName")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/component_recommendations.go:15]: Check PK uniqueness, should add `account_id` and `region`?

File: plugins/source/aws/resources/services/resiliencehub/sop_recommendations.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func sopRecommendations() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_sop_recommendations",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_SopRecommendation.html`,
+		Resolver:    fetchSopRecommendations,
+		Transform:   transformers.TransformWithStruct(&types.SopRecommendation{}, transformers.WithPrimaryKeys("RecommendationId")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/sop_recommendations.go:15]: Check PK uniqueness if the recommendations have global IDs or not

File: plugins/source/aws/resources/services/resiliencehub/test_recommendations.go [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+func testRecommendations() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_resiliencehub_test_recommendations",
+		Description: `https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_TestRecommendation.html`,
+		Resolver:    fetchTestRecommendations,
+		Transform:   transformers.TransformWithStruct(&types.TestRecommendation{}, transformers.WithPrimaryKeys("ReferenceId")),
```
Comment (disq) [inline plugins/source/aws/resources/services/resiliencehub/test_recommendations.go:15]: Check PK uniqueness

**Review by disq — CHANGES_REQUESTED** [review]

**Comment by disq** [comment]
`PhysicalResourceId` is hopefully unique but with names and others I'm not so sure.

**Comment by disq** [comment]
Nice fix, also added PK to `arn` fields, which could be unneeded... I quickly checked a few resources in `aws/` and if the `arn` is the PK we don't seem to add `account_id`/`region` as additional PKs.

File: plugins/source/aws/docs/tables/aws_resiliencehub_apps.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
+# Table: aws_resiliencehub_apps
+
+https://docs.aws.amazon.com/resilience-hub/latest/APIReference/API_App.html
+
+The composite primary key for this table is (**account_id**, **region**, **arn**).
```
Comment (hermanschaaf) [inline plugins/source/aws/docs/tables/aws_resiliencehub_apps.md:5]: Assuming the `arn` contains account ID and region components, we should be able to drop account ID and region from the composite PK and only keep `arn`.

**Review by hermanschaaf — APPROVED** [review]
LGTM, probably just that one PK that could be simplified a little

**Review by disq — APPROVED** [review]
